warehouse/accounts/utils.py

```diff
@@ -44,3 +51,17 @@ class UserContext:
 
     def __principals__(self) -> list[str]:
         return self.user.__principals__()
+
+
+def update_email_domain_status(email: Email, request: Request) -> None:
+    """
+    Update the domain status of the given email address.
+    """
+    domain_status_service = request.find_service(IDomainStatusService)
+    domain_status = domain_status_service.get_domain_status(email.domain)
```

Comment:
There's a failure case here where this returns `[]`:

https://github.com/pypi/warehouse/blob/a685531499e81433727bb51e7a0ef9db2ffbf0d2/warehouse/accounts/services.py#L1007-L1009

I think we shouldn't consider a domain for which we weren't able to get a status 'checked', and it should probably go back into the pool of unchecked domains to be retried, unless I'm missing something we will just store this empty status, which will remain until 30 days have passed.

Reply:
Thanks - I've updated the logic and included a test case to confirm that we'll re-check these before the threshold is reached, as we won't update the record at all.